src/js/components/Button/Button.js

```diff
@@ -281,8 +310,17 @@ const Button = forwardRef(
             theme.global.colors.text[isDarkBackground() ? 'dark' : 'light'],
         });
       }
+    } else if (kind && kindIcon && !plain) {
+      const iconColor =
+        (hover && getIconColor(themePaths.hover, theme)) ||
```

Comment:
This gets the color of the icon correctly. However, the way we have the icon defined in the theme:
```
icon: <FormNext />
```
when adding hover[kind].icon, my expectation would be that I could specify another icon for the hover state:
```
hover: {
  ...
  icon: <Add color="blue" />
  ...
}
```
that my icon would also change, but only the color swaps.

Do we want to support both? Or should we rethink how the icon property is structured?

Reply:
I would think the icon would also change -- let me look into this.

Reply:
Updated in recent commit:

https://user-images.githubusercontent.com/12522275/186539652-b734b80b-3780-4e50-b583-ff58b3d5df59.mov

Reply:
👍